demisto_sdk/commands/common/clients/__init__.py

```diff
@@ -112,14 +128,31 @@ def get_client_from_server_type(
     Returns:
         the correct client based on querying the type of the server
     """
+    _base_url = base_url or os.getenv(DEMISTO_BASE_URL, "")
+    _api_key = api_key or os.getenv(DEMISTO_KEY, "")
+    _auth_id = auth_id or os.getenv(AUTH_ID)
+    _username = username or os.getenv(DEMISTO_USERNAME, "")
+    _password = password or os.getenv(DEMISTO_PASSWORD, "")
+    _verify_ssl = (
+        verify_ssl
+        if verify_ssl is not None
+        else string_to_bool(os.getenv(DEMISTO_VERIFY_SSL, False))
+    )
+
     _client = demisto_client.configure(
-        base_url=base_url,
-        api_key=api_key,
-        verify_ssl=verify_ssl,
-        username=username,
-        password=password,
+        base_url=_base_url,
+        api_key=_api_key,
+        auth_id=_auth_id,
+        verify_ssl=_verify_ssl,
+        username=_username,
+        password=_password,
     )
 
+    if not XsoarClient.is_xsoar_healthy(_client):
```

Comment:
we should have this check happen based on a init param
check_server_health
with a default True

Reply:
done